### ISSUE TITLE
False positive on private vars using with-redefs

### ISSUE THREAD

**Issue by ITSOES** [borkdude/clj-kondo#678]
Using `with-redefs` with a private variable of another namespace throws a “is private” linting error.

E.g.
```
echo '
(ns bar)
(defn- private-name [] "John")
(defn public-hello []
  (str "Hello, " (private-name)))

(ns foo (:require [bar :as barr]))
(defn -main []
  (with-redefs [barr/private-name (fn [] "Jeff")]
    (barr/public-hello)))
' | clj-kondo --lint -
```
returns:
```
<stdin>:11:17: error: #'bar/private-name is private
linting took 47ms, errors: 1, warnings: 0
```

There shouldn’t be any error with redefining a private var using `(with-redefs)`.

**Comment by borkdude** [comment]
Thanks.
